src/Microsoft.Extensions.Configuration.AzureAppConfiguration/ConfigurationClientManager.cs

```diff
@@ -21,31 +23,56 @@ namespace Microsoft.Extensions.Configuration.AzureAppConfiguration
     /// there won't be multiple threads calling this client at the same time.
     /// </remarks>
     internal class ConfigurationClientManager : IConfigurationClientManager
-    {
-        private readonly IList<ConfigurationClientWrapper> _clients;
-
-        public ConfigurationClientManager(IEnumerable<string> connectionStrings, ConfigurationClientOptions clientOptions)
-        {
-            if (connectionStrings == null || !connectionStrings.Any())
-            {
-                throw new ArgumentNullException(nameof(connectionStrings));
-            }
-
-            _clients = connectionStrings.Select(connectionString => 
-            {
-                var endpoint = new Uri(ConnectionStringParser.Parse(connectionString, ConnectionStringParser.EndpointSection));
-                return new ConfigurationClientWrapper(endpoint, new ConfigurationClient(connectionString, clientOptions));
-            }).ToList();
-        }
-
-        public ConfigurationClientManager(IEnumerable<Uri> endpoints, TokenCredential credential, ConfigurationClientOptions clientOptions)
-        {
-            if (endpoints == null || !endpoints.Any())
-            {
-                throw new ArgumentNullException(nameof(endpoints));
-            }
-
-            _clients = endpoints.Select(endpoint => new ConfigurationClientWrapper(endpoint, new ConfigurationClient(endpoint, credential, clientOptions))).ToList();
+    {
+        private IList<ConfigurationClientWrapper> _clients;
+
+        private IEnumerable<string> _connectionStrings;
+        private IEnumerable<Uri> _endpoints;
+        private TokenCredential _credential;
+        private ConfigurationClientOptions _clientOptions;
+
+        public ConfigurationClientManager(AzureAppConfigurationOptions options)
+        {
+            if (options == null)
+            {
+                throw new ArgumentNullException(nameof(options));
+            }
+
+            if (options.ClientOptions == null)
+            {
+                throw new ArgumentNullException(nameof(options.ClientOptions));

```

Comment:
This should be ArgumentException, since its technically not the argument that is null - one of its properties is null.